src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1227,6 +1257,52 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 					reflectionContext.RecordUnrecognizedPattern (2058, $"Parameters passed to method '{calledMethodDefinition.GetDisplayName ()}' cannot be analyzed. Consider using methods 'System.Type.GetType' and `System.Activator.CreateInstance` instead.");
 					break;
 
+				//
+				// System.Reflection.Assembly
+				//
+				// Location ()
+				//
+				case IntrinsicId.Assembly_Location when _context.SingleFileAnalysis == true:
+					_context.LogWarning ($"'{calledMethodDefinition.GetDisplayName ()}' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.", 3000, callingMethodDefinition);
+					break;
+
+				//
+				// System.Reflection.Assembly
+				//
+				// CodeBase ()
+				//
+				case IntrinsicId.Assembly_CodeBase when _context.SingleFileAnalysis == true:
+					_context.LogWarning ($"'{calledMethodDefinition.GetDisplayName ()}' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.", 3000, callingMethodDefinition);
```

Comment:
This will actually throw, not return a empty string. I think we should add new warnings for this and other APIs (other than Assembly.Location and Assembly.GetFiles).

Reply:
And now that we have these warnings being produced by the linker, we should probably add them to the docs and close https://github.com/mono/linker/issues/1498.